### DISCUSSION TITLE
css-file für BSC

### DISCUSSION THREAD

**Discussion by WagnerSystems** [shining-man/bsc_fw#26] [Show and tell]
Hallo,
ich habe den Code etwas umgebaut so das css-files vom spiffs Datei-System geladen werden können.
Falls Interesse besteht kann ich das gerne teilen.
![Home](https://user-images.githubusercontent.com/23194134/219943270-7cbb5427-aac0-4d9b-a330-0a9c9a359f21.png)

Es hat mich auch gestört das der GitHub-Link im gleichen Tab geöffnet wurde, den Link habe ich auf ein neues Tab umgeleitet.
`"<div class='card' onclick=\"window.open('https://github.com/shining-man/bsc_fw','_blank');\"><p>BSC on GitHub</p></div>"`
@Tobias das soll keine Kritik darstellen, deine Arbeit ist super. Vielen Dank das du das geteilt hast.

**Comment by shining-man** [comment]
Ich betrachte dies nicht als Kritik. Ich freue mich wenn es anderen hilft und dabei noch weiter wächst. Wenn ich das nicht wollte, hätte ich es nicht veröffentlicht.
Du kannst es gerne teilen, dann schaue ich mir es einmal an.

Der Grund warum ich dies bisher noch nicht gemacht habe ist eingentlich das instabile SPIFFS. Vielleicht gibt es hier aber auch Abhilfe die einfach übersehen habe.
Das Logging leite ich in ein Logfile um. Kommt es jetzt zu einem Spannungsausfall wird unter Umständen das Log-File beschädigt. Dies führt dazu, dass es nicht mehr vollsätndig geladen werden kann. Was man merkt, wenn man dieses über den "Log"-Button anschauen möchte, da der ESP dann nie mit dem Laden fertig wird. Das beschädigte Logfile lässt sich auch nicht mehr löschen. Die einzige Abhilfe ist die Formatierung der Partition. Damit ist aber dann leider der ganze Inhalt weg.
Ich habe als Alternative LittleFS ausprobiert. Da gibt es dies Problem nicht, aber die Stabilität ist nicht so toll.

Hast du hier evtl. Erfahrung?

**Comment by WagnerSystems** [comment]
Hier mal ein diff von meinen Änderungen. Musste die Datei als txt benennen weil patch nicht angenommen wurde als Datei Endung.
[css.txt](https://github.com/shining-man/bsc_fw/files/10782003/css.txt)
